jujugui/static/gui/src/app/models/handlers.js

```diff
@@ -110,6 +110,76 @@ YUI.add('juju-delta-handlers', function(Y) {
         result.tags = tags.join(',');
       }
       return result;
+    },
+
+    /**
+      Translates the new JujuStatus and WorkloadStatus's to the legacy Juju 1
+      values.
+
+      This logic is a JS implementation of the Juju core implementation to
+      accomplish the same task:
+      https://github.com/
+        juju/juju/blob/juju-1.26-alpha1/state/status_model.go#L272
+
+      @method translateToLegacyAgentState
+      @param {String} agentStatus JujuStatus.Current
+      @param {String} workloadStatus WorkloadStatus.Current
+      @param {String} workloadStatusMessage WorkloadStatus.Message
+    */
+    translateToLegacyAgentState: function(
+      agentStatus, workloadStatus, workloadStatusMessage) {
+      var statusMaintenance = 'maintenance';
+      var statusAllocating = 'allocating';
+      var statusPending = 'pending';
+      var statusError = 'error';
+      var statusRebooting = 'rebooting';
+      var statusExecuting = 'executing';
+      var statusIdle = 'idle';
+      var statusLost = 'lost';
+      var statusFailed = 'failed';
+      var statusTerminated = 'terminated';
+      var statusStarted = 'started';
+      var statusStopped = 'stopped';
+
+      var messageInstalling = 'installing charm software';
+      var isInstalled = workloadStatus != statusMaintenance ||
+                        workloadStatusMessage != messageInstalling;
+
+      switch (agentStatus) {
+
+        case statusAllocating:
+          return statusPending;
+          break;
+
+        case statusError:
+          return statusError;
+          break;
+
+        case statusRebooting:
+        case statusExecuting:
+        case statusIdle:
+        case statusLost:
+        case statusFailed:
+          switch (workloadStatus) {
+
+            case statusError:
+              return statusError;
+              break;
+
+            case statusTerminated:
+              return statusStopped;
+              break;
+
+            case statusMaintenance:
+              return isInstalled ? statusStarted : statusPending;
```

Comment:
Since in this case workloadStatus is always statusMaintenance, the isInstalled condition can be only met if workloadStatusMessage != messageInstalling. Or maybe I am missing something. If not, then we could get rid of isInstalled and use:
```javascript
return workloadStatusMessage === messageInstalling ? statusPending : statusStarted;
```